doc/dev/osd_internals/mclock_wpq_cmp_study.rst

```diff
@@ -0,0 +1,488 @@
+=========================================
+ QoS Study with mClock and WPQ Schedulers
+=========================================
+
+Ceph's use of mClock was primarily experimental and approached with an
+exploratory mindset. This is still true with other organizations and individuals
+who continue to either use the codebase or modify it according to their needs.
+
+DmClock exists in its own repository_. Before the Ceph *Pacific* release,
+mClock could be enabled by setting the ``osd_op_queue`` Ceph option to
+*mclock_scheduler*. Additional mClock parameters like *reservation*, *weight*
+and *limit* for each service type could be set using Ceph options.
+For example, ``osd_mclock_scheduler_client_[res,wgt,lim]`` is one such option.
+See `OSD Config Reference`_ section for more details. Even with all the mClock
```

Comment:
better off using
```rst
:ref:`dmclock-qos`
```

the relative link is fragile.

Reply:
Sure, I will push this in my next update.